lib/application.js

```diff
@@ -567,3 +571,65 @@ app.listen = function(cb) {
 
   return server;
 };
+
+/* START CONTROLLER CODE*/
+
+// this is similar to app.model
+// at the moment I haven't really used 'options'
+app.controller = function(ctor, options) {
+  var self = this;
+
+  // the SharedController is a wrapper, see below
+  var controller = new SharedController(ctor);
+  this.remotes().addClass(controller.sharedClass);
+  app.controllers().push(controller);
+
+  return controller;
+
+  // SharedController class contains strong-remoting objects to set up the metadata
+  // and register the endpoint, etc
+  function SharedController(ctor, options) {
+    options = options || {};
+    this.name = ctor.name;
+    this.ctor = ctor;
+    // TODO - right now there's a hack on line 289 to skip this
+    // util.inherits(ctor, EventEmitter);
+
+    this.sharedClass = new RemoteObjects.SharedClass(this.name, ctor, options);
+    // the resolveScope method instantiates the controller upon request
+    // (please correct my terminology if it's wrong)
+    this.sharedClass.resolveScope = function(remotingContext) {
+      // self is the app - should it be something more obvious like appInstance?
+      return new ctor(self, remotingContext);
+    };
+
+    // similar to model.remoteMethod
+    this.remoteMethod = function(name, options) {
+      // I didn't dive into why, but this only worked when I set isStatic to true
+      // whether the controller methods were prototype methods or not
+      options.isStatic = true;
```

Comment:
Let me explain. When `isStatic:true`, strong-remoting will call the method on the scope provided by `resolveScope` function. When `isStatic:false`, strong-remoting will use the scope as a constructor function and call the method on an instance created by this constructor. (Obviously, that assumes that the scope is a constructor function, which is true for LoopBack models, but it's not true for controllers).

Because SharedController returns a controller instance as the scope, all remote methods defined on a controller must be marked as static, even though they are defined on a prototype like instance methods. I admit this may be a little bit confusing.

Recently, @0candy was changing the API for working with remote methods, see https://github.com/strongloop/loopback/issues/597 and https://github.com/strongloop-internal/scrum-loopback/issues/617. I am could not quickly find a comprehensive overview of our discussions, here is the essence: Instead of identifying methods by name+isStatic flag, we encode this flag in the method name. For example, a static method is "find" and a prototype method is "prototype.updateAttributes". See RELEASE NOTES in [loopback](https://github.com/strongloop/loopback/blob/363bc4d6c182067cf02fdf964ee4ea38ce408f35/3.0-RELEASE-NOTES.md#new-method-of-defining-remoting-metadata) and [strong-remoting](https://github.com/strongloop/strong-remoting/blob/aa391d2e65460196ae2a59493f9dd4844622f35d/3.0-RELEASE-NOTES.md#invocation-for-remote-method-has-been-changed).

>```js
{ 
methods: {
  staticMethod: {
    http: { path: '/static' }
  },
  'prototype.instanceMethod': {
  http: { path: '/instance' }
}}
```

I think this is fortunate for controllers, because there is no `isStatic` flag around to confuse users. It also means that you should not need to set it here, as long as you are using the latest strong-remoting version from the master.